### DISCUSSION TITLE
Map Generation: Color in Height Maps to apply terain texture

### DISCUSSION THREAD

**Discussion by leicestersquare** [OpenRCT2/OpenRCT2#20982] [Feature requests]
The use of height maps is very effective for creating complex terrain topography.
As for now, it only maps greyscale to height of map tiles.

A colour to tile texture conversion would be a great addition to the generator.
An intuitive and backwards compatible way would be to use hue, saturation and luminance to define texture, terrain edge and height.
Prefering HSL allows better control with common tools available in most image editing software.

The chosen texture would not need to be evenly distributed across the hue spectrum - it could be roughly similar to the terrain colour and take up the available scale acording to its relevance to make it intuitive. (e.g. large spectrum of grass in green and small spectrum of neon-terain in pink).

**Comment by xrgtmx** [comment]
Excellent idea. I also propose that despite what color scheme is set per height, I think that having a function that will automatically fill a color in by height range would be nice, too. This function could be used on heightmaps _and_ simplex generated maps.
